aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/ObserveQueryManager.java

```diff
@@ -0,0 +1,219 @@
+package com.amplifyframework.datastore.storage.sqlite;
+
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.core.Action;
+import com.amplifyframework.core.Consumer;
+import com.amplifyframework.core.async.Cancelable;
+import com.amplifyframework.core.model.Model;
+import com.amplifyframework.core.model.query.QueryOptions;
+import com.amplifyframework.datastore.DataStoreConfiguration;
+import com.amplifyframework.datastore.DataStoreException;
+import com.amplifyframework.datastore.DataStoreItemChange;
+import com.amplifyframework.datastore.DataStoreQuerySnapshot;
+import com.amplifyframework.datastore.storage.ItemChangeMapper;
+import com.amplifyframework.datastore.storage.StorageItemChange;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import io.reactivex.rxjava3.disposables.Disposable;
+import io.reactivex.rxjava3.subjects.Subject;
+
+
+public class ObserveQueryManager<T extends Model> implements Cancelable {
```

Comment:
Avoid name "manager," it usually doesn't have any meaning. In _what way_ does this component manage the Observe query? By answering that question, you can probably come up with a more accurate descriptor.